src/test/java/org/example/servletsHomework/servlet/functionalTests/FunctionalTextServletTest.java

```diff
@@ -0,0 +1,337 @@
+
+        String expectedOutput = "Text not found with id " + textId;
+        assertEquals(expectedOutput, response.body());
+        assertEquals(HttpServletResponse.SC_NOT_FOUND, response.statusCode());
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenInvalidPath() throws URISyntaxException, IOException, InterruptedException {
+
+        List<String> invalidPaths = List.of("/afds", "/!", "/@@", "/1/2/3/");
+
+        for (String path : invalidPaths) {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI(baseUrl + textUrl + path))
+                    .GET()
+                    .header("token", token)
+                    .build();
+
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+            String expectedOutput = "Invalid request path: " + path;
+            assertEquals(expectedOutput, response.body());
+            assertEquals(HttpServletResponse.SC_BAD_REQUEST, response.statusCode());
+        }
+    }
+
+    @Test
+    void shouldPostTextThenGetById() throws URISyntaxException, IOException, InterruptedException {
+        String jsonRequest = """
+                {
+                "text": "TestText123"
+                }
+                """;
+
+        HttpRequest requestPost = HttpRequest.newBuilder()
+                .uri(new URI(baseUrl + textUrl))
+                .POST(HttpRequest.BodyPublishers.ofString(jsonRequest))
+                .header("token", token)
+                .header("Content-Type", "application/json")
+                .build();
+
+        HttpResponse<String> responsePost = client.send(requestPost, HttpResponse.BodyHandlers.ofString());
+        assertEquals(HttpServletResponse.SC_CREATED, responsePost.statusCode());
+
+        String responseBody = responsePost.body();
+
+        String messageStart = "\"textId\":";
+        int idStartIndex = responseBody.indexOf(messageStart) + messageStart.length();
+        int idEndIndex = responseBody.indexOf("}", idStartIndex);
+        String textId = responseBody.substring(idStartIndex, idEndIndex).trim();
+
+        assertTrue(responseBody.contains("\"text\":\"TestText123\""));
+        assertTrue(responseBody.contains("\"message\":\"Text saved with id: " + textId));
+
+        HttpRequest requestGet = HttpRequest.newBuilder()
+                .uri(new URI(baseUrl + textUrl + textId))
+                .GET()
+                .header("token", token)
+                .build();
+
+        HttpResponse<String> responseGet = client.send(requestGet, HttpResponse.BodyHandlers.ofString());
+
+        String expectedOutput = "Text with id " + textId + ": TestText123";
+        assertEquals(expectedOutput, responseGet.body());
+        assertEquals(HttpServletResponse.SC_OK, responseGet.statusCode());
+
+    }
+
+    @Test
+    void shouldPostTextThenDeleteById() throws URISyntaxException, IOException, InterruptedException {
```

Comment:
shouldDeteletTextById. This is case you are trying to test. Text creation it is pre-action which is required to test your case here.